charts/k8s-monitoring/templates/agent_config/_profiles_pprof.river.txt

```diff
@@ -0,0 +1,162 @@
+{{ define "agent.config.profilesPprof" }}
+// Profiles
+{{- if .Values.profiles.ebpf.enabled }}
+discovery.kubernetes "pprof_pods" {
+  selectors {
+    role = "pod"
+    field = "spec.nodeName=" + env("HOSTNAME")
+  }
+{{- if .Values.profiles.pprof.namespaces }}
+  namespaces {
+    names = {{ .Values.profiles.pprof.namespaces | toJson }}
+  }
+{{- end }}
+  role = "pod"
+}
+
+discovery.relabel "pprof_pods" {
+  targets = concat(discovery.kubernetes.pprof_pods.targets)
+
+  rule {
+    action        = "drop"
+    source_labels = ["__meta_kubernetes_pod_phase"]
+    regex         = "Pending|Succeeded|Failed|Completed"
+  }
+
+  rule {
+    action = "labelmap"
+    regex  = "__meta_kubernetes_pod_label_(.+)"
+  }
+
+  rule {
+    action        = "replace"
+    source_labels = ["__meta_kubernetes_namespace"]
+    target_label  = "namespace"
+  }
+
+  rule {
+    action        = "replace"
+    source_labels = ["__meta_kubernetes_pod_name"]
+    target_label  = "pod"
+  }
+
+  rule {
+    action        = "replace"
+    source_labels = ["__meta_kubernetes_pod_container_name"]
+    target_label  = "container"
+  }
+{{- if .Values.profiles.pprof.extraRelabelingRules }}
+{{ .Values.profiles.pprof.extraRelabelingRules | indent 2 }}
+{{- end }}
+}
+
+{{- $profileTypes := .Values.profiles.pprof.types }}
+{{ range $profileTypes }}
+discovery.relabel "pprof_pods_{{.}}_default_name" {
```

Comment:
Just for my own understanding, what does this syntax mean: `{{.}}`

Reply:
Typically. `.` is the top level data object. so `.Values` means delve inside the values `.Release` goes into Helm's pre-defined release object.

The `range` command iterates over a list and resets `.` to each item in that list. That's also why sometimes you see saving a variable before the range, as in with
```
$profileTypes := .Values.profiles.pprof.types
```
because that way we can still reference that item even though `.` has become something else.


You'll also see things like
```
{{ with .Values.externalServices.prometheus }}
```
which resets `.` to be starting at `.Values.externalServices.prometheus`.

Reply:
In this case, `{{ . }}` is referencing the current element in the `.Values.profiles.pprof.types` array